testsuite/sysconfig-get1.cc

```diff
@@ -42,4 +43,29 @@ BOOST_AUTO_TEST_CASE(sysconfig_get1)
     BOOST_CHECK_EQUAL(boost::join(tmp_vector, "-"), "c:\\io.sys");
 
     BOOST_CHECK(!s.getValue("V5", tmp_vector));
+
+    BOOST_CHECK(s.getValue("V6", tmp_vector));
+    BOOST_CHECK_EQUAL(boost::join(tmp_vector, "-"), "a-value-with-a-#-hash");
+}
+
+BOOST_AUTO_TEST_CASE(sysconfig_set1)
+{
+    system("cp sysconfig-set1.txt sysconfig-set1.txt.tmp");
+    SysconfigFile s("sysconfig-set1.txt.tmp");
+
+    string tmp_string;
+
+    BOOST_CHECK(s.getValue("K2", tmp_string));
+    BOOST_CHECK_EQUAL(tmp_string, "changeme");
+
+    s.setValue("K2", "all new");
+    BOOST_CHECK(s.getValue("K2", tmp_string));
+    BOOST_CHECK_EQUAL(tmp_string, "all new");
+
+    s.setValue("K2", "changeme");
+    BOOST_CHECK(s.getValue("K2", tmp_string));
+    BOOST_CHECK_EQUAL(tmp_string, "changeme");
+
+    s.save();
+    BOOST_CHECK_EQUAL(system("diff -u sysconfig-set1.txt sysconfig-set1.txt.tmp"), 0);
```

Comment:
NP: That `.tmp` file is marked in .gitignore, but maybe we could delete it if the test succeeds, I do not like leaving garbage on the disk...

Reply:
It should also be possible to make the comparison in memory.

Reply:
I tried to do it without writing the temporary file, by using AsciiFile::lines and then BOOST_CHECK_EQUAL_COLLECTIONS, but SysconfigFile is a *protected* subclass of AsciiFile so that method is inaccessible. @aschnell did you have another way in mind?

Reply:
OK, lines() can indeed not be used right away. So some extension to SysConfigFile would be needed.